packages/ripple-binary-codec/src/types/st-object.ts

```diff
@@ -125,8 +127,11 @@ class STObject extends SerializedType {
       const associatedValue = field.associatedType.from(
         xAddressDecoded[field.name],
       )
-
-      bytes.writeFieldAndValue(field, associatedValue)
+      if ((associatedValue as unknown as Bytes).name === 'UNLModify') {
+        isUnlModify = true
+      }
```

Comment:
nbd but could just be

```javascript
const isUnlModify = (associatedValue as unknown as Bytes).name === 'UNLModify'
```

Reply:
This actually doesn't work - it's not this field that requires the `isUnlModify`, it's the `Account` field that needs to know this

Reply:
woof.

Reply:
you might want to comment that since this could be a refactoring trap later

Reply:
The way that works definitely took me a while to understand